pynicotine/portmapper.py

```diff
@@ -126,17 +127,25 @@ def _request_port_mapping(self, public_port, private_port, lease_duration):
 
         with socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM, proto=socket.IPPROTO_UDP) as sock:
             sock.bind((self.local_ip_address, 0))
-            sock.settimeout(self.REQUEST_TIMEOUT)
 
             request = self.PortmapRequest(public_port, private_port, lease_duration)
-            request.sendto(sock, (self._gateway_address, self.REQUEST_PORT))
 
-            try:
-                response = self.PortmapResponse(message=sock.recv(16))
-                return response.result
+            timeout = self.REQUEST_INIT_TIMEOUT
+            i = 1
+            while i <= self.REQUEST_ATTEMPTS:
+                sock.settimeout(timeout)
+                request.sendto(sock, (self._gateway_address, self.REQUEST_PORT))
+
+                try:
+                    response = self.PortmapResponse(message=sock.recv(16))
+                    return response.result
+
+                except socket.timeout:
+                    timeout *= 2
+                    log.add_debug(f"NAT-PMP: Attempt {i} timed out, increased timeout to {timeout}s")
+                    i += 1
 
-            except socket.timeout:
-                pass
+        log.add_debug(f"NAT-PMP: Fatal: all attempts timed out")
```

Comment:
`f"NAT-PMP: Giving up, all {self.REQUEST_ATTEMPTS} attempts to map the port timed out"`

This might be more friendly to help troubleshooting end users who may not know what NAT-PMP is.